### PR TITLE
New CaloL{1,2} LUTs are taken care of by the L1T O2O

### PR DESCRIPTION
**Pull request by kkotov**
This PR supersedes PR #18872.
It includes recent development of CaloL2 including tauTrimmingVeto LUT and several commits for removing  Stage2 from CaloParams record names.

### CODE REVIEW

**Comment by franzoni** [comment]
please test

**Comment by kkotov** [comment]
@franzoni , thanks for triggering the test! And thanks to @arunhep for testing the original #18872 although in the end it fell out of synch with the cms-l1t-offline.

**Comment by kkotov** [comment]
@rekovic , could you sign-off no this PR?

**Comment by rekovic** [comment]
+1

**Comment by kkotov** [comment]
@ggovi , could you review and sign-off this small PR? From the AlCa/DB point of view it only adds a new record: CondFormats/DataRecord/interface/L1TCaloParamsO2ORcd.h

**Comment by ggovi** [comment]
+1

**Comment by arunhep** [comment]
+1

File: L1TriggerConfig/L1TConfigProducers/src/CaloParamsHelperO2O.h [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+
+    // regions
+    double regionLsb() const { return regionLsb_; }
+    std::string regionPUSType() const { return pnode_[regionPUS].type_; }
+    std::vector<double> regionPUSParams() { return pnode_[regionPUS].dparams_; }
+    l1t::LUT* regionPUSLUT() {return &pnode_[regionPUS].LUT_; }
+
+    int regionPUSValue(int PUM0, int eta)
+    {
+      int puSub = ceil(regionPUSParams()[18*eta+PUM0]*2);
```
Comment (davidlange6) [inline L1TriggerConfig/L1TConfigProducers/src/CaloParamsHelperO2O.h]: hi @kkotov - please document the meaning of 18 in a followup pr. thanks

**Comment by davidlange6** [comment]
+1

**Pull request merged**